lib/perl/Genome/InstrumentData/AlignmentResult/WithIntermediateResults.pm

```diff
@@ -63,10 +76,20 @@ sub get_or_create_intermediate_result_for_params {
 sub remove_intemediate_results {
     my $self = shift;
 
+    my $result_users = $self->_user_data_for_nested_results;
+
     my @iar_users = Genome::SoftwareResult::User->get(user => $self, label => intermediate_result_user_label);
     for my $iar_user ( @iar_users ) {
         my $iar = $iar_user->software_result;
         $iar_user->active(0);
+
+        if($result_users) {
+            for my $value (values %$result_users) {
+                my @using = $iar->users(user => $value, label => ['created', 'shortcut', 'sponsor']);
+                map $_->active(0), @using;
```

Comment:
`delete()` won't necessarily know which user relationships are safe to mark as inactive.  We're selectively marking specifically our own, but `delete()` will fail if any others remain.  This is an addition to the `active(0)` being set on line 84.